src/vm/costs/mod.rs

```diff
@@ -194,6 +260,235 @@ pub enum CostErrors {
     CostContractLoadFailure,
 }
 
+fn load_state_summary(clarity_db: &mut ClarityDatabase) -> Result<CostStateSummary> {
+    let last_processed_at = match clarity_db.get_value(
+        "vm-costs::last-processed-at-height",
+        &TypeSignature::UIntType,
+    ) {
+        Some(v) => u32::try_from(v.expect_u128()).expect("Block height overflowed u32"),
+        None => return Ok(CostStateSummary::empty()),
+    };
+
+    let metadata_result = clarity_db
+        .fetch_metadata_manual::<String>(
+            last_processed_at,
+            &STACKS_BOOT_COST_VOTE_CONTRACT,
+            "state_summary",
+        )
+        .map_err(|e| CostErrors::CostComputationFailed(e.to_string()))?;
+    let serialized: SerializedCostStateSummary = match metadata_result {
+        Some(serialized) => serde_json::from_str(&serialized).unwrap(),
+        None => return Ok(CostStateSummary::empty()),
+    };
+    Ok(CostStateSummary::from(serialized))
+}
+
+fn store_state_summary(
+    clarity_db: &mut ClarityDatabase,
+    to_store: &CostStateSummary,
+) -> Result<()> {
+    let block_height = clarity_db.get_current_block_height();
+
+    clarity_db.put(
+        "vm-costs::last-processed-at-height",
+        &Value::UInt(block_height as u128),
+    );
+    let serialized_summary =
+        serde_json::to_string(&SerializedCostStateSummary::from(to_store.clone()))
+            .expect("BUG: failure to serialize cost state summary struct");
+    clarity_db.set_metadata(
+        &STACKS_BOOT_COST_VOTE_CONTRACT,
+        "state_summary",
```

Comment:
This key can never conflict with anything the cost voting contract will ever insert, right?  Would it make sense to make this value an illegal Clarity string, or something, to make sure?

Reply:
Yeah, it couldn't conflict with anything that the contract would insert (metadata keys aren't contract chosen), but I can add some more illegal characters to be sure.

Reply:
Added in ee998d20dc043c9cac68669aa8f03a7a8a02f407